packages/contracts-bedrock/test/L2/L2ToL2CrossDomainMessenger.t.sol

```diff
@@ -0,0 +1,415 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+// Testing utilities
+import { Test } from "forge-std/Test.sol";
+import { Reverter } from "test/mocks/Callers.sol";
+
+// Libraries
+import { Predeploys } from "src/libraries/Predeploys.sol";
+
+// Target contract
+import {
+    L2ToL2CrossDomainMessenger,
+    NotEntered,
+    MessageDestinationSameChain,
+    RelayCallerNotCrossL2Inbox,
+    CrossL2InboxOriginNotL2ToL2CrossDomainMessenger,
+    MessageDestinationNotRelayChain,
+    MessageTargetCrossL2Inbox,
+    MessageTargetL2ToL2CrossDomainMessenger,
+    MessageAlreadyRelayed
+} from "src/L2/L2ToL2CrossDomainMessenger.sol";
+import { CrossL2Inbox } from "src/L2/CrossL2Inbox.sol";
+
+/// @title L2ToL2CrossDomainMessengerTest
+/// @dev The L2ToL2CrossDomainMessengerTest contract tests the L2ToL2CrossDomainMessenger contract.
+contract L2ToL2CrossDomainMessengerTest is Test {
+    /// @dev L2ToL2CrossDomainMessenger contract instance.
+    L2ToL2CrossDomainMessenger l2ToL2CrossDomainMessenger;
+
+    /// @dev Sets up the test suite.
+    function setUp() public {
+        // Deploy the L2ToL2CrossDomainMessenger contract
+        vm.etch(Predeploys.CROSS_L2_INBOX, address(new CrossL2Inbox()).code);
```

Comment:
You can use [graphite](https://graphite.dev/) to more easily manage stacked PRs, like this PR depends on the previous PR and graphite is a tool that makes it easy to work on PRs that depend on each other, it automates rebasing, so you could have the cross l2 inbox code as part of this PR and any times you change the cross l2 inbox this PR could be automatically updated

Reply:
good idea!